### ISSUE TITLE
dvc exp run --queue does not use local credentials on config.local

### ISSUE THREAD

**Issue by samtzai** [iterative/dvc#7567]
# Bug Report 

## Issue name
exp: dvc exp run does not use credentials stored on config.local when queueing a experiment

## Description
This issue happens when you queue an experiment with dvc>=2.9.4 and then you try to reproduce it with dvc exp run --run-all

### Reproduce
* Get a project with a working dvc repro pipeline.
* Create a project remote with credentials on .dvc/config and .dvc/condig.local:
```console
	dvc remote add my_remote my_remote_address
	dvc remote modify --local my_remote connection_string my_connection_string
```
* queue an experiment
```console
	dvc exp run --queue --set-param model.sense_of_life=31
```
* run an experiment
```console
	dvc exp run --run-all
```

Error when applied to a azure remote:
```console
	ERROR: failed to reproduce 'train': Authentication to Azure Blob Storage requires either account_name or connection_string.
```
Error when applied to a http remote:
```console
	ERROR: failed to reproduce 'train': config file error: HTTP 'custom' authentication require both 'custom_auth_header' and 'password'
```
However, if we execute dvc repro, it executes correctly. If we execute the experiment with dvc<2.9.4 it executes correctly.

### Expected
Correct experiment execution as it happens with dvc<2.9.4

### Environment information

**Output of `dvc doctor`:**
```console
$ dvc doctor
DVC version: 2.10.1 (pip)
---------------------------------
Platform: Python 3.8.10 on Linux-5.4.0-104-generic-x86_64-with-glibc2.29
Supports:
        azure (adlfs = 2022.2.0, knack = 0.9.0, azure-identity = 1.9.0),
        webhdfs (fsspec = 2022.2.0),
        http (aiohttp = 3.8.1, aiohttp-retry = 2.4.6),
        https (aiohttp = 3.8.1, aiohttp-retry = 2.4.6)
Cache types: hardlink, symlink
Cache directory: ext4 on /dev/md126
Caches: local
Remotes: https, azure
Workspace directory: ext4 on /dev/md126
Repo: dvc, git
```

**Additional Information (if any):**

**Comment by dberenbaum** [comment]
@pmrowla Is this a duplicate of #7254? If so, is it expected that this is a regression that started with 2.9.4?

**Comment by daavoo** [comment]
Closing on favor of #7254

**Comment by samtzai** [comment]
BTW, I have realized that if you set a fake password or connection_string on the config file it works fine.

**Comment by dberenbaum** [comment]
Reopening to figure out what's going on and whether this is truly a duplicate.

@samtzai How were you using azure during the experiments? Did you have external dependencies or outputs in your experiment?

> BTW, I have realized that if you set a fake password or connection_string on the config file it works fine.

Could you explain what you mean?

**Comment by samtzai** [comment]
Hi @dberenbaum, 

Maybe this clarifies. 
No external dependencies, just dependencies from dataset already pushed into azure before experiment. Actually the remote is not relevant.  

- What I meant is that if you have this .dvc/config:

`[core]
    remote = my_remote
['remote "my_remote"']
    url = azure://my_azure/
    connection_string =fake_connection_string_to_make_experiments_work`
and this .dvc/config.local:
`['remote "my_remote"']
 connection_string =real_connection_string`

and you run dvc exp:

`dvc exp run --queue --set-param model.sense_of_life=32 -n my_exp
dvc exp run --run-all
dvc exp push origin my_exp`

It works perfectly and even if able to push the experiment.

- However, if you have this .dvc/config:

`[core]
    remote = my_remote
['remote "my_remote"']
    url = azure://my_azure/
and this .dvc/config.local:
`['remote "my_remote"']
 connection_string =real_connection_string`

and you run dvc exp:

`dvc exp run --queue --set-param model.sense_of_life=32 -n my_exp
dvc exp run --run-all
dvc exp push origin my_exp`

you get the following error: 

`ERROR: failed to reproduce 'train': Authentication to Azure Blob Storage requires either account_name or connection_string.`

THis means that there is a checking somewhere  (Maybe a Schema?) that requires .dvc/config to be complete (connection_string for azure, password for https for example). However, if you set a fake value for this key on the config, when executes the experiment it is actually using the value from the config.local (as it is correctly pushing the experiment with dvc exp push

I have uploaded a dvc.yaml and a experiments.sh file that reproduces this error:
[issue_7567.zip](https://github.com/iterative/dvc/files/8476821/issue_7567.zip)

**Comment by pmrowla** [comment]
Can you run (without the dummy `connection_string`)
```
dvc exp run --temp --set-param model.sense_of_life=32 -n my_exp -v
```
and post the full error traceback here

**Comment by samtzai** [comment]
**Experiment run with dvc==2.10.1 with the fake password:**
```console
> python3 dummy_test.py --metrics outputs
2022-04-19 12:30:25,793 DEBUG: state save (198532216, 1650364225729406720, 69) 786701871bd46b1f3f228a7acb4ac4f7
2022-04-19 12:30:25,836 DEBUG: state save (198532216, 1650364225729406720, 69) 786701871bd46b1f3f228a7acb4ac4f7
2022-04-19 12:30:25,836 DEBUG: Output 'outputs/metrics.json' doesn't use cache. Skipping saving.
2022-04-19 12:30:25,838 DEBUG: Computed stage: 'test' md5: '33e31b1b232f147c393a4cc7b1981e15'
2022-04-19 12:30:25,883 DEBUG: state save (198532216, 1650364225729406720, 69) 786701871bd46b1f3f228a7acb4ac4f7                                                                                                   
2022-04-19 12:30:25,884 DEBUG: Preparing to transfer data from 'memory://dvc-staging/2d62fd40b06a6c7f0bc78c147862251311f87089d2c207f2b03765582ce37bcf' to '/mnt/RAID5/users/samtzai/projects/experiments_addon/.dvc/cache'
2022-04-19 12:30:25,884 DEBUG: Preparing to collect status from '/mnt/RAID5/users/samtzai/projects/experiments_addon/.dvc/cache'
2022-04-19 12:30:25,884 DEBUG: Collecting status from '/mnt/RAID5/users/samtzai/projects/experiments_addon/.dvc/cache'
2022-04-19 12:30:25,885 DEBUG: Preparing to collect status from 'memory://dvc-staging/2d62fd40b06a6c7f0bc78c147862251311f87089d2c207f2b03765582ce37bcf'                                                           
2022-04-19 12:30:25,888 DEBUG: state save (198532216, 1650364225729406720, 69) 786701871bd46b1f3f228a7acb4ac4f7                                                                                                   
2022-04-19 12:30:25,892 DEBUG: Uploading '/mnt/RAID5/users/samtzai/projects/experiments_addon/.dvc/cache/.eqa3Zmad2WSaXtiGa4kFrv.tmp' to '/mnt/RAID5/users/samtzai/projects/experiments_addon/.dvc/tmp/exps/tmpw__qaidk/outputs/.8fbHvrW23swFa6Vp2sJaYv.tmp'
2022-04-19 12:30:25,895 DEBUG: Removing '/mnt/RAID5/users/samtzai/projects/experiments_addon/.dvc/tmp/exps/tmpw__qaidk/outputs/.8fbHvrW23swFa6Vp2sJaYv.tmp'                                                       
2022-04-19 12:30:25,895 DEBUG: Removing '/mnt/RAID5/users/samtzai/projects/experiments_addon/.dvc/cache/.eqa3Zmad2WSaXtiGa4kFrv.tmp'
2022-04-19 12:30:25,895 DEBUG: Removing '/mnt/RAID5/users/samtzai/projects/experiments_addon/.dvc/tmp/exps/tmpw__qaidk/outputs/metrics.json'
2022-04-19 12:30:25,896 DEBUG: Uploading '/mnt/RAID5/users/samtzai/projects/experiments_addon/.dvc/cache/78/6701871bd46b1f3f228a7acb4ac4f7' to '/mnt/RAID5/users/samtzai/projects/experiments_addon/.dvc/tmp/exps/tmpw__qaidk/outputs/metrics.json'
2022-04-19 12:30:25,897 DEBUG: state save (198532221, 1650364225893404416, 69) 786701871bd46b1f3f228a7acb4ac4f7                                                                                                   
2022-04-19 12:30:25,970 DEBUG: state save (198532221, 1650364225893404416, 69) 786701871bd46b1f3f228a7acb4ac4f7
2022-04-19 12:30:25,979 DEBUG: stage: 'test' was reproduced
Updating lock file 'dvc.lock'
2022-04-19 12:30:25,992 DEBUG: Staging files: {'outputs/metrics.json', 'dvc.yaml', 'params.yaml', 'dvc.lock'}
2022-04-19 12:30:26,199 DEBUG: Commit to new experiment branch 'refs/exps/73/ecae965629c109ff4136d9d2bc0b49a158a4af/my_exp_2'
2022-04-19 12:30:26,682 DEBUG: Collected experiment '5b4498a'.
2022-04-19 12:30:26,683 DEBUG: Removing tmpdir '/mnt/RAID5/users/samtzai/projects/experiments_addon/.dvc/tmp/exps/tmpw__qaidk'
2022-04-19 12:30:26,683 DEBUG: Removing '/mnt/RAID5/users/samtzai/projects/experiments_addon/.dvc/tmp/exps/tmpw__qaidk'
2022-04-19 12:30:26,686 DEBUG: Removing '/mnt/RAID5/users/samtzai/projects/experiments_addon/.dvc/tmp/exps/run/4592c25ab73570f58c576aa58819833edaa78fe3'
                                                                      
Reproduced experiment(s): my_exp_2
To apply the results of an experiment to your workspace run:

        dvc exp apply <exp>

To promote an experiment to a Git branch run:

        dvc exp branch <exp> <branch>

2022-04-19 12:30:26,709 DEBUG: Analytics is enabled.
2022-04-19 12:30:26,761 DEBUG: Trying to spawn '['daemon', '-q', 'analytics', '/tmp/tmp6ash95gf']'
2022-04-19 12:30:26,763 DEBUG: Spawned '['daemon', '-q', 'analytics', '/tmp/tmp6ash95gf']'
```

**Experiment run with dvc==2.10.1 without the fake password in .dvc/config:**
```console
-----------------------------------------------------------
2022-04-19 12:36:20,389 DEBUG: Analytics is enabled.
2022-04-19 12:36:20,416 DEBUG: Trying to spawn '['daemon', '-q', 'analytics', '/tmp/tmpkemzzjn0']'
2022-04-19 12:36:20,417 DEBUG: Spawned '['daemon', '-q', 'analytics', '/tmp/tmpkemzzjn0']'
(env) samtzai@yann:~/projects/experiments_addon$ dvc exp run --temp --set-param model.sense_of_life=50 -n my_exp_5 -v
2022-04-19 12:36:27,751 DEBUG: Using experiment params 'defaultdict(<class 'dict'>, {'params.yaml': {'model.sense_of_life': 50}})'
2022-04-19 12:36:28,117 DEBUG: Stashed experiment 'd7f4b1e' with baseline '73ecae9' for future execution.
2022-04-19 12:36:28,127 DEBUG: Reproducing experiment revs 'd7f4b1e'
2022-04-19 12:36:28,504 DEBUG: Writing experiments local config '/mnt/RAID5/users/samtzai/projects/experiments_addon/.dvc/tmp/exps/tmpx9s_3v4d/.dvc/config.local'
2022-04-19 12:36:28,504 DEBUG: Init temp dir executor in '/mnt/RAID5/users/samtzai/projects/experiments_addon/.dvc/tmp/exps/tmpx9s_3v4d'
2022-04-19 12:36:29,028 DEBUG: Running repro in '/mnt/RAID5/users/samtzai/projects/experiments_addon/.dvc/tmp/exps/tmpx9s_3v4d'
2022-04-19 12:36:29,029 DEBUG: Removing '/mnt/RAID5/users/samtzai/projects/experiments_addon/.dvc/tmp/exps/tmpx9s_3v4d/.dvc/tmp/repro.dat'
2022-04-19 12:36:29,143 DEBUG: Dependency 'params.yaml' of stage: 'test' changed because it is '{'model': 'modified'}'.
2022-04-19 12:36:29,144 DEBUG: stage: 'test' changed.
2022-04-19 12:36:29,145 DEBUG: Removing output 'outputs/metrics.json' of stage: 'test'.
2022-04-19 12:36:29,145 DEBUG: Removing '/mnt/RAID5/users/samtzai/projects/experiments_addon/.dvc/tmp/exps/tmpx9s_3v4d/outputs/metrics.json'
2022-04-19 12:36:29,167 DEBUG: defaultdict(<class 'dict'>, {'params.yaml': {'model': 'modified'}})
Running stage 'test':
> python3 dummy_test.py --metrics outputs
2022-04-19 12:36:29,237 DEBUG: state save (198532219, 1650364589216458496, 69) 1ff832e2af5ac0c92bf4a00768c7bf8a
2022-04-19 12:36:29,263 DEBUG: state save (198532219, 1650364589216458496, 69) 1ff832e2af5ac0c92bf4a00768c7bf8a
2022-04-19 12:36:29,263 DEBUG: Output 'outputs/metrics.json' doesn't use cache. Skipping saving.
2022-04-19 12:36:29,265 DEBUG: Computed stage: 'test' md5: '4d0ca35d7970999af2208fa6424f55b3'
2022-04-19 12:36:29,276 DEBUG: state save (198532219, 1650364589216458496, 69) 1ff832e2af5ac0c92bf4a00768c7bf8a                                                                                                   
2022-04-19 12:36:29,277 DEBUG: Preparing to transfer data from 'memory://dvc-staging/2d62fd40b06a6c7f0bc78c147862251311f87089d2c207f2b03765582ce37bcf' to '/mnt/RAID5/users/samtzai/projects/experiments_addon/.dvc/cache'
2022-04-19 12:36:29,277 DEBUG: Preparing to collect status from '/mnt/RAID5/users/samtzai/projects/experiments_addon/.dvc/cache'
2022-04-19 12:36:29,277 DEBUG: Collecting status from '/mnt/RAID5/users/samtzai/projects/experiments_addon/.dvc/cache'
2022-04-19 12:36:29,277 DEBUG: Preparing to collect status from 'memory://dvc-staging/2d62fd40b06a6c7f0bc78c147862251311f87089d2c207f2b03765582ce37bcf'                                                           
2022-04-19 12:36:29,279 DEBUG: state save (198532219, 1650364589216458496, 69) 1ff832e2af5ac0c92bf4a00768c7bf8a                                                                                                   
2022-04-19 12:36:29,285 DEBUG: Uploading '/mnt/RAID5/users/samtzai/projects/experiments_addon/.dvc/cache/.RRb9mirEBFFFaFNPHGT2Be.tmp' to '/mnt/RAID5/users/samtzai/projects/experiments_addon/.dvc/tmp/exps/tmpx9s_3v4d/outputs/.bqXgmbhKmKA7cNjSABGRBn.tmp'
2022-04-19 12:36:29,287 DEBUG: Removing '/mnt/RAID5/users/samtzai/projects/experiments_addon/.dvc/tmp/exps/tmpx9s_3v4d/outputs/.bqXgmbhKmKA7cNjSABGRBn.tmp'                                                       
2022-04-19 12:36:29,288 DEBUG: Removing '/mnt/RAID5/users/samtzai/projects/experiments_addon/.dvc/cache/.RRb9mirEBFFFaFNPHGT2Be.tmp'
2022-04-19 12:36:29,288 DEBUG: Removing '/mnt/RAID5/users/samtzai/projects/experiments_addon/.dvc/tmp/exps/tmpx9s_3v4d/outputs/metrics.json'
2022-04-19 12:36:29,289 DEBUG: Uploading '/mnt/RAID5/users/samtzai/projects/experiments_addon/.dvc/cache/1f/f832e2af5ac0c92bf4a00768c7bf8a' to '/mnt/RAID5/users/samtzai/projects/experiments_addon/.dvc/tmp/exps/tmpx9s_3v4d/outputs/metrics.json'
2022-04-19 12:36:29,290 DEBUG: state save (198532227, 1650364589284457728, 69) 1ff832e2af5ac0c92bf4a00768c7bf8a                                                                                                   
2022-04-19 12:36:29,374 DEBUG: state save (198532227, 1650364589284457728, 69) 1ff832e2af5ac0c92bf4a00768c7bf8a
2022-04-19 12:36:29,384 DEBUG: stage: 'test' was reproduced
Updating lock file 'dvc.lock'
2022-04-19 12:36:29,398 DEBUG: Staging files: {'params.yaml', 'outputs/metrics.json', 'dvc.lock', 'dvc.yaml'}
2022-04-19 12:36:29,588 DEBUG: Commit to new experiment branch 'refs/exps/73/ecae965629c109ff4136d9d2bc0b49a158a4af/my_exp_5'
2022-04-19 12:36:30,050 DEBUG: Collected experiment '621f9a3'.
2022-04-19 12:36:30,051 DEBUG: Removing tmpdir '/mnt/RAID5/users/samtzai/projects/experiments_addon/.dvc/tmp/exps/tmpx9s_3v4d'
2022-04-19 12:36:30,051 DEBUG: Removing '/mnt/RAID5/users/samtzai/projects/experiments_addon/.dvc/tmp/exps/tmpx9s_3v4d'
2022-04-19 12:36:30,056 DEBUG: Removing '/mnt/RAID5/users/samtzai/projects/experiments_addon/.dvc/tmp/exps/run/d7f4b1e696940797bd93b89eed5f2e54070a90b2'
                                                                      
Reproduced experiment(s): my_exp_5
To apply the results of an experiment to your workspace run:

        dvc exp apply <exp>

To promote an experiment to a Git branch run:

        dvc exp branch <exp> <branch>

2022-04-19 12:36:30,082 DEBUG: Analytics is enabled.
2022-04-19 12:36:30,134 DEBUG: Trying to spawn '['daemon', '-q', 'analytics', '/tmp/tmpx47c38oi']'
2022-04-19 12:36:30,136 DEBUG: Spawned '['daemon', '-q', 'analytics', '/tmp/tmpx47c38oi']'
```
Both executions work.

**Comment by pmrowla** [comment]
@samtzai so this issue is resolved for you in 2.10.1?

**Comment by samtzai** [comment]
I have to say that I am not able to reproduce the issues I got last week even if I created these dummy project to recreate it and it failed last week. Apparently, nothing has changed on the enviroment and the version, this dummy code has no external dependencies and thus is, reproductable. However, I am not able to replicate the error now, even with dvc==3.10.0.
According to dvc doctor the relevant library versions are the same. 

So, I would say that misteriously the issue is apparently solved. Probably the issue can be closed and I if find something coherent I can post it here.

DVC version: 2.10.0 (pip)
---------------------------------
Platform: Python 3.8.10 on Linux-5.4.0-104-generic-x86_64-with-glibc2.29
Supports:
        azure (adlfs = 2022.2.0, knack = 0.9.0, azure-identity = 1.9.0),
        webhdfs (fsspec = 2022.2.0),
        http (aiohttp = 3.8.1, aiohttp-retry = 2.4.6),
        https (aiohttp = 3.8.1, aiohttp-retry = 2.4.6)
Cache types: hardlink, symlink
Cache directory: ext4 on /dev/md126
Caches: local
Remotes: https, azure
Workspace directory: ext4 on /dev/md126
Repo: dvc, git
